executor/executor.go

```diff
@@ -536,6 +536,14 @@ func (e *executor[T]) runBlocks(params Params, processor Processor[T], extension
 	return err
 }
 
+func PreRun[T any](params Params, extensions []Extension[T]) (err error) {
+	state := State[T]{}
+	ctx := Context{State: params.State}
+
+	state.Block = params.From
+	return signalPreRun(state, &ctx, extensions)
```

Comment:
For completeness, I recommend supporting both opening and closing of a scope -- i.e., `signalPreRun` and `signalPostRun`.

Reply:
Thank you, that makes sense